samcli/lib/test_runner/fargate_testsuite_runner.py

```diff
@@ -0,0 +1,472 @@
+            self.color.yellow(
+                f"=> Compressing and uploading {self.tests_path} to {Path(bucket).joinpath(self.path_in_bucket).as_posix()}\n"
+            )
+        )
+
+        # Compress tests into a temporary tarfile to send to S3 bucket
+        with tarfile.open(temp_tarfile_name, "w:gz") as tar:
+            tar.add(self.tests_path, arcname=self.tests_path.stem)
+
+        # Upload compressed file to S3 bucket
+        try:
+            with open(temp_tarfile_name, "rb") as tests_tar:
+                self.boto_s3_client.put_object(
+                    Body=tests_tar,
+                    Bucket=bucket,
+                    Key=self.path_in_bucket.joinpath(Path(self.COMPRESSED_TESTS_FILE_NAME)).as_posix(),
+                )
+        finally:
+            # Remove the tarfile after the upload is complete
+            os.remove(temp_tarfile_name)
+
+        LOG.info(
+            self.color.yellow(
+                f"=> Uploading {self.requirements_file_path} to {Path(bucket).joinpath(self.path_in_bucket).as_posix()}\n"
+            )
+        )
+
+        with open(self.requirements_file_path, "rb") as requirements_file:
+            self.boto_s3_client.put_object(
+                Body=requirements_file,
+                Bucket=bucket,
+                Key=self.path_in_bucket.joinpath(Path(self.REQUIREMENTS_FILE_NAME)).as_posix(),
+            )
+
+        waiter = self.boto_s3_client.get_waiter("object_exists")
+        waiter.wait(Bucket=bucket, Key=self.path_in_bucket.joinpath(Path(self.COMPRESSED_TESTS_FILE_NAME)).as_posix())
+        waiter.wait(Bucket=bucket, Key=self.path_in_bucket.joinpath(Path(self.REQUIREMENTS_FILE_NAME)).as_posix())
+
+        LOG.info(self.color.green("✓ Tests and requirements sucessfully uploaded, kicking off testsuite...\n"))
+
+    def _download_results(self, bucket: str) -> None:
+        """
+        Downloads test results from the bucket, as a tarball. The tarball is expanded, and removed after expansion.
+
+        Parameters
+        ----------
+        bucket : str
+            The bucket from which test results are downloaded.
+
+            This is passed in because it is not known at the time of FargateTestsuiteRunner creation where the bucket name will come from.
+            If a bucket override is not provided, the bucket contained in the Test Runner Stack must be used. The stack must be created/updated before this
+            bucket can be fetched.
+        """
+        self.boto_s3_client.download_file(
+            Bucket=bucket,
+            Key=self.path_in_bucket.joinpath(self.COMPRESSED_RESULTS_FILE_NAME).as_posix(),
+            Filename=self.COMPRESSED_RESULTS_FILE_NAME,
+        )
+
+        LOG.info(
+            self.color.green(f"✓ Results sucessfully downloaded and saved into {self.COMPRESSED_RESULTS_FILE_NAME}  \n")
+        )
+
+        with tarfile.open(self.COMPRESSED_RESULTS_FILE_NAME) as results_tar:
+            results_tar.extractall(".")
```

Comment:
Should the output go into a sub directory? I'm not sure if folks running this will appreciate the root of their project getting populated with test output.

Reply:
The output is already enclosed in a sub-directory, this should definitely be commented. The Fargate container will package results into a directory and then tar that directory.